### PR TITLE
feat(kno-2628): add trigger_data param support

### PR DESCRIPTION
**Pull request by juanazam**
Add support for the `trigger_data` param on the following endpoints:

* Messages list
* Message activities
* User messages
* Object messages
* User feed

### CODE REVIEW

File: lib/knock/resources/helpers.ex [hunk 1]
```diff
@@ -0,0 +1,16 @@
+defmodule Knock.Helpers do
+  def maybe_encode_trigger_data(options) do
+    case options[:trigger_data] do
+      trigger_data when is_map(trigger_data) ->
+        trigger_data
+        |> Jason.encode!()
+        |> then(&Keyword.put(options, :trigger_data, &1))
```
Comment (cjbell) [inline lib/knock/resources/helpers.ex:7]: What version of elixir do we need to pin to to support then?

File: lib/knock/resources/helpers.ex [hunk 2]
```diff
@@ -0,0 +1,16 @@
+defmodule Knock.Helpers do
+  def maybe_encode_trigger_data(options) do
```
Comment (cjbell) [inline lib/knock/resources/helpers.ex:2]: Docs and specs please! We also might want to put this in a more specific helpers module, or make this more generic as an encode json data as option and pass in the name of the key

**Review by cjbell — COMMENTED** [review]
2 comments

File: lib/knock/resources/helpers.ex (hunk 2 above)
Comment (juanazam) [reply lib/knock/resources/helpers.ex:2]: Will make it more generic!

File: lib/knock/resources/helpers.ex (hunk 1 above)
Comment (juanazam) [reply lib/knock/resources/helpers.ex:7]: Will remove this `then` usage so we can go to lower versions!

**Review by cjbell — APPROVED** [review]
💯

**Pull request merged**